### ISSUE TITLE
TODO: Documentation of 'begin/end with support'

### ISSUE THREAD

**Issue by KinoxKlark** [pyimgui/pyimgui#265]
As suggested in #264, this new functionality should be properly documented.

**Comment by mCodingLLC** [comment]
Do we want docs to have both old and new style? If so should we present the new style or the old style first in the documentation?

**Comment by KinoxKlark** [comment]
I think both should be documented, and in order to nudge the good practice, we should start with the new one. Maybe with a structure like:
* Here is how it is done using the `with` keyword
* In order to be consistent with the original API it is also possible to do it with `begin`/`end` pair
Those are just some thoughts, I am open to any suggestions.

**Comment by KinoxKlark** [comment]
Yeah, documentation is kind of the elephant in the room that desperately needs to be addressed but that I keep procrastinating ^^. As you may know, I am not the original creator and maintainer of this project and I just recently took the lead, I still have some things to figure out. 

As far as I understand it the documentation is automatically built from doctstrings with special blocs that are parsed and used for the example display (`.. visual-example::`), the reference to DearImGui (`.. wraps::`), and maybe some more. This means that the one `imgui.core` page is an artifact of the fact that the `core` module is in one file, and to be fair, I don't know how we can change that (but I didn't take much time to inspect). 

More importantly, the automated build of the documentation via readthedocs is broken since the project changed of hands and I still have to take some time to fix it.

In the meantime, I would suggest to properly documenting via docstrings. I'll fix the built for sure before releasing 2.0, which would automatically include properly documented features to the documentation. If someone sees an easy way for splitting into sub-pages I am open to it. It would indeed make sense to segment a bit the documentation and to make it more readable.
